src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -1887,6 +1888,136 @@ public static void XmlSerializerFactoryTest()
         Assert.Equal(dog1.Breed, dog2.Breed);
     }
 
+    [Fact]
+    public static void XmlUnknownElementAndEventHandlerTest()
+    {
+        List<string> grouplists = new List<string>();
+        int count = 0;
+        XmlSerializer serializer = new XmlSerializer(typeof(Group));
+        serializer.UnknownElement += new XmlElementEventHandler((o, args) =>
+        {
+            Group myGroup = (Group)args.ObjectBeingDeserialized;
+            Assert.NotNull(myGroup);
+            grouplists.Add(args.Element.Name);
+            ++count;
+        });
+        string xmlFileContent = @"<?xml version=""1.0"" encoding=""utf-8""?>
+  <Group xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xmlns:xsd = ""http://www.w3.org/2001/XMLSchema"">
+         <GroupName>MyGroup</GroupName>
+         <GroupSize>Large</GroupSize>
+         <GroupNumber>444</GroupNumber>
+         <GroupBase>West</GroupBase>
+       </Group >";
+        Group group = (Group)serializer.Deserialize(GetStreamFromString(xmlFileContent));
+        Assert.NotNull(group);
+        Assert.NotNull(group.GroupName);
+        Assert.Null(group.GroupVehicle);
+        Assert.Equal(3, count);
+        Assert.Equal(3, grouplists.Count());
+        bool b = grouplists.Contains("GroupSize") && grouplists.Contains("GroupNumber") && grouplists.Contains("GroupBase");
+        Assert.True(b);
+    }
+
+    [Fact]
+    public static void XmlUnknownNodeAndEventHandlerTest()
+    {
+        List<string> grouplists = new List<string>();
+        int count = 0;
+        XmlSerializer serializer = new XmlSerializer(typeof(Group));
+        serializer.UnknownNode += new XmlNodeEventHandler((o, args) =>
+        {
+            Group myGroup = (Group)args.ObjectBeingDeserialized;
+            Assert.NotNull(myGroup);
+            grouplists.Add(args.LocalName);
+            ++count;
+        });
+        string xmlFileContent = @"<?xml version=""1.0"" encoding=""utf-8""?>
+  <Group xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xmlns:xsd=""http://www.w3.org/2001/XMLSchema"" xmlns:coho=""http://www.cohowinery.com"" xmlns:cp=""http://www.cpandl.com"">
+              <coho:GroupName>MyGroup</coho:GroupName>
+              <cp:GroupSize>Large</cp:GroupSize>
+              <cp:GroupNumber>444</cp:GroupNumber>
+              <coho:GroupBase>West</coho:GroupBase>
+              <coho:ThingInfo>
+                    <Number>1</Number>
+                    <Name>Thing1</Name>
+                    <Elmo>
+                        <Glue>element</Glue>
+                    </Elmo>
+              </coho:ThingInfo>
```

Comment:
Can we change the unknown nodes' names to "UnknownNode1", "UnknownNode2", and etc.?